frontend/src/components/organisms/ManageUserProfile.tsx

```diff
@@ -13,23 +13,186 @@ import LinearProgress from "@mui/material/LinearProgress";
 import Link from "next/link";
 import ArrowBackIosNewIcon from "@mui/icons-material/ArrowBackIosNew";
 import { IconButton } from "@mui/material";
+import { BASE_URL } from "@/utils/constants";
+import { auth } from "@/utils/firebase";
+import Alert from "../atoms/Alert";
+import { useRouter } from "next/router";
+import { Grid } from "@mui/material";
+import { useAuth } from "@/utils/AuthContext";
+import Modal from "@/components/molecules/Modal";
 
-interface ManageUserProfileProps {
+type userProfileData = {
+  name: string;
+  role: string;
+  email: string;
+  joinDate: string;
   userid: string;
+  hours: number;
+  status: string;
+};
+
+interface ManageUserProfileProps {
+  userProfileDetails: userProfileData;
 }
 
+type userStatusData = {
+  userRole: string;
+  userStatus: string;
+  userID: string;
+};
+
+type userRegistrationData = {
+  totalHours: number;
+  userRegistrations: any[];
+};
+
+type verifyData = {
+  totalHours: number;
+};
+
+function formatDateString(dateString: string) {
+  const date = new Date(dateString);
+  const month = (date.getMonth() + 1).toString().padStart(2, "0");
+  const day = date.getDate().toString().padStart(2, "0");
+  const year = date.getFullYear();
+  return `${month}/${day}/${year}`;
+}
+
+const ModalBody = ({
```

Comment:
Come to think of it, do you think we can we keep the fetch logic inside the `Status` component and only use the ModalBody for "content"